text/0000-else.md

```diff
@@ -0,0 +1,219 @@
+
+## Basic example
+
+```html
+<!-- component.html -->
+<template>
+    <template if:true={abra}>
+        Abra!
+    </template>
+    <template elseif:true={kadabra}>
+        Kadabra!
+    </template>
+    <template lwc:else>
+        Alakazam!
+    </template>
+</template>
+```
+
+## Motivation
+
+As of today, without "else" or "else if" directives, the developer has to chain `if:true` and `if:false` blocks together:
+
+```html
+<template>
+    <template if:true={abra}>
+        Abra!
+    </template>
+    <template if:false={abra}>
+        <template if:true={kadabra}>
+            Kadabra!
+        </template>
+        <template if:false={kadabra}>
+            Alakazam!
+        </template>
+    </template>
+</template>
+```
+
+This is awkward and unintuitive, and has an additional performance cost in that the property getters 
+(in this case, `abra` and `kadabra`) are called multiple times, when once should suffice.
+
+### Prior art
+
+The concept of "else" and "else if" exists in most programming languages, and in many other frameworks:
+
+- [Vue: `v-else` and `v-else-if`](https://vuejs.org/guide/essentials/conditional.html#v-else)
+- [Svelte: `:else` and `:else if`](https://svelte.dev/docs#template-syntax-if)
+- [Angular: `ngIfElse`](https://angular.io/api/common/NgIf)
+- [Solid: `<Show>` and `<Match>`](https://www.solidjs.com/docs/latest/api#%3Cshow%3E)
+- [Aura: `aura:set attribute="else"`](https://developer.salesforce.com/docs/component-library/bundle/aura:if/documentation)
+
+Some other frameworks, such as [React](https://reactjs.org/docs/conditional-rendering.html), [Stencil](https://stenciljs.com/docs/templating-jsx#conditionals), and [Lit](https://lit.dev/docs/templates/conditionals/#conditionals-with-if-statements), have no built-in concept of `else` or `else if`, other than that provided by JavaScript itself.
+
+In the case of LWC, since we already have `if:else` and `if:false`, and since there is no ergonomic way to do conditional rendering in JavaScript (other than `render()`, which requires creating separate template HTML files), it would be most natural to align with the first group of frameworks.
+
+## Detailed design
+
+The basic design is to add three new directives:
+
+- `elseif:true`
+- `elseif:false`
+- `lwc:else`
+
+The reason for using `lwc:else` rather than just `else` is to avoid conflicts with existing attributes that may be named `else`, and to ensure that all directives have a namespace separator (`:`).
+
+### Syntax
+
+All three new directives must be immediately preceded by a sibling `if:true`/`if:false` (or `elseif:true`/`elseif:false`).
+
+```html
+<!-- Valid -->
+<template if:true={foo}></template>
+<template lwc:else></template>
+```
+
+```html
+<!-- Invalid -->
+<template if:true={foo}></template>
+<div>Yolo!</div>
+<template lwc:else></template>
+```
+
+Whitespace is ignored when considering siblinghood.

```

Comment:
I would add a comment here, stating that `if:*`, `esleif:*` and `lwc:else` are mutually exclusive on the same element.

Reply:
Done.